chromeos.kernelci.org

```diff
@@ -0,0 +1,224 @@
+#!/bin/sh
+
+# Periodic job run on staging.kernelci.org to merge together all the pending
+# PRs and update the staging.kernelci.org branches, then update Docker images
+# and trigger a full build/test cycle with a kernel branch based on linux-next.
+
+set -e
+
+CHROMEOS_SETTINGS="data/chromeos.ini"
+
+cmd_pull() {
+    echo "Updating local repository"
+    git pull --ff-only
+}
+
+cmd_jenkins() {
+    echo "Updating Jenkins jobs"
+    ./pending.py kernelci-jenkins --settings=${CHROMEOS_SETTINGS} --main=chromeos.kernelci.org --push
+
+    [ -d checkout/kernelci-jenkins/data ] || {
+        # To set this up:
+        # git clone https://github.com/kernelci/kernelci-jenkins-data.git data
+        # cd data; git-crypt unlock
+        # cd ..; ln -s data/bot.staging.kernelci.org/env .env
+
+        echo "Encrypted kernelci-jenkins-data repository not set up"
+        return 1
+    }
+
+    echo "Updating encrypted data checkout"
+    topdir=$(pwd)
+    cd checkout/kernelci-jenkins/data
+    git remote update origin
+    data_diff=$(git diff origin/main)
+    if [ -n "$data_diff" ]; then
+        git checkout origin/main
+        cd "$topdir/checkout/kernelci-jenkins"
+        echo "Recreating Jenkins container"
+        docker-compose down
+        docker-compose up --build -d
+        sleep 60  # magic - another way to do this would be to poll something
+    else
+        cd "$topdir/checkout/kernelci-jenkins"
+        # ToDo: restart the container but only when there were some changes in
+        # kernelci-jenkins to avoid aborting running jobs unnecessarily
+    fi
+    cd "$topdir"
+
+    echo "Triggering Jenkins seed job"
+    python3 job.py --no-params trigger job-dsl
+}
+
+cmd_core() {
+    echo "Updating kernelci-core"
+    ./pending.py kernelci-core --settings=${CHROMEOS_SETTINGS} --main=chromeos.kernelci.org --push
+}
+
+cmd_bootrr() {
+    echo "Updating bootrr"
+    ./pending.py bootrr --settings=${CHROMEOS_SETTINGS} --main=chromeos.kernelci.org --push
+}
+
+cmd_buildroot() {
+    echo "Updating buildroot"
+    ./pending.py buildroot --settings=${CHROMEOS_SETTINGS} --main=chromeos.kernelci.org --push
+}
+
+cmd_test_definitions() {
+    echo "Updating test-definitions"
+    ./pending.py test-definitions --settings=${CHROMEOS_SETTINGS} --main=chromeos.kernelci.org --push
+}
+
+cmd_backend() {
+    opt="$1"
+
+    echo "Checking for missing indexes by finding COLLSCAN events"
+    echo "db.system.profile.aggregate([{\$match: {ns: {\$in: ['chromeos-kernel-ci.test_case', 'chromeos-kernel-ci.test_group', 'chromeos-kernel-ci.test_regression', 'chromeos-kernel-ci.build']}, planSummary: 'COLLSCAN'}}, {\$group: {_id: {ns: '\$ns', planSummary: '\$planSummary'}}}])" | mongo chromeos-kernel-ci
+
+    if [ "$opt" = "indexes" ]; then
+        echo "Dropping Mongo DB indexes"
+        echo "db.test_case.dropIndexes(); db.test_group.dropIndexes(); db.test_regression.dropIndexes(); db.build.dropIndexes();" | mongo chromeos-kernel-ci
+    fi
```

Comment:
Please drop this, we might only enable it if we start making changes to the schema used for chromeos.kernelci.org.